src/rgw/rgw_aio.h

```diff
@@ -79,6 +85,15 @@ class Aio {
 
   // wait for all outstanding completions and return their results
   virtual AioResultList drain() = 0;
+
+  // Use unique pointer here because while the current interface may
+  // wait on a condition variable, future non-blocking versions may
+  // well *not* call `aio_operate` before returning, so
+  // ObjectOperations on the stack are unsafe.
+  static OpFunc librados_op(librados::ObjectReadOperation&& op,
+			    optional_yield y = null_yield);
+  static OpFunc librados_op(librados::ObjectWriteOperation&& op,
+			    optional_yield y = null_yield);
```

Comment:
nit: would prefer not to hide `null_yield` behind a default argument. if we force callers to document that they're using this synchronously, it will be easier to audit the code paths that we expect to be fully-async